commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/PrincipalToGroup.java

```diff
@@ -0,0 +1,115 @@
+package org.wildfly.extras.creaper.commands.security.realms;
+
+/**
+ * Helper class which holds information principal-to-group group search in ldap authorization of security realm.
+ */
+public final class PrincipalToGroup {
+
+    private final String groupAttribute;
+    private final String groupDnAttribute;
+    private final String groupName;
+    private final String groupNameAttribute;
+    private final Boolean iterative;
+    private final Boolean preferOriginalConnection;
+    private final Boolean skipMissingGroups;
+    private final LdapCache cache;
+
+    private PrincipalToGroup(Builder builder) {
+        this.groupAttribute = builder.groupAttribute;
+        this.groupDnAttribute = builder.groupDnAttribute;
+        this.groupNameAttribute = builder.groupNameAttribute;
+        this.groupName = builder.groupName;
+        this.iterative = builder.iterative;
+        this.preferOriginalConnection = builder.preferOriginalConnection;
+        this.skipMissingGroups = builder.skipMissingGroups;
+        this.cache = builder.cache;
+    }
+
+    public String getGroupAttribute() {
+        return groupAttribute;
+    }
+
+    public String getGroupDnAttribute() {
+        return groupDnAttribute;
+    }
+
+    public String getGroupName() {
+        return groupName;
+    }
+
+    public String getGroupNameAttribute() {
+        return groupNameAttribute;
+    }
+
+    public Boolean getIterative() {
+        return iterative;
+    }
+
+    public Boolean getPreferOriginalConnection() {
+        return preferOriginalConnection;
+    }
+
+    public Boolean getSkipMissingGroups() {
+        return skipMissingGroups;
+    }
+
+    public LdapCache getCache() {
+        return cache;
+    }
```

Comment:
Do the methods have to be `public`? I mean, is the class meant to be used for other things than creating a command? If it's only for creating the command, the methods can be omitted and the `private` variables made package-protected.